src/server/scripts/Spells/spell_dh.cpp

```diff
@@ -16,11 +16,142 @@
+            }
+        };
+        AuraScript* GetAuraScript() const override
+        {
+            return new spell_dh_chaos_strike_AuraScript();
+        }
+};
+
+// 195072 - Fel Rush
+class spell_dh_fel_rush : public SpellScriptLoader
+{
+    public:
+        spell_dh_fel_rush() : SpellScriptLoader("spell_dh_fel_rush") { }
+
+        class spell_dh_fel_rush_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_dh_fel_rush_SpellScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_FEL_RUSH_GROUND))
+                    return false;
+                if (!sSpellMgr->GetSpellInfo(SPELL_FEL_RUSH_AIR))
+                    return false;
+                return true;
+            }
+
+            void HandleDashGround(SpellEffIndex /*effIndex*/)
+            {
+                if (Unit* caster = GetCaster())
+                    if (!caster->IsFalling() || caster->IsInWater())
+                    {
+                        caster->CastSpell(caster, SPELL_FEL_RUSH_GROUND, true);
+                        caster->CastSpell(caster, SPELL_FEL_RUSH_DAMAGE, true);
+                    }
+            }
+
+            void HandleDashAir(SpellEffIndex /*effIndex*/)
+            {
+                if (Unit* caster = GetCaster())
+                    if (caster->IsFalling())
+                    {
+                        caster->SetDisableGravity(true);
+                        caster->CastSpell(caster, SPELL_FEL_RUSH_AIR, true);
+                        caster->CastSpell(caster, SPELL_FEL_RUSH_DAMAGE, true);
+                    }
+            }
+
+            void Register() override
+            {
+                OnEffectHitTarget += SpellEffectFn(spell_dh_fel_rush_SpellScript::HandleDashGround, EFFECT_0, SPELL_EFFECT_DUMMY);
+                OnEffectHitTarget += SpellEffectFn(spell_dh_fel_rush_SpellScript::HandleDashAir, EFFECT_1, SPELL_EFFECT_DUMMY);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_dh_fel_rush_SpellScript();
+        }
+};
+
+// 197923 - Fel Rush (air cast)
+class spell_dh_fel_rush_aura : public SpellScriptLoader
+{
+    public:
+        spell_dh_fel_rush_aura() : SpellScriptLoader("spell_dh_fel_rush_aura") { }
+
+        class spell_dh_fel_rush_aura_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_dh_fel_rush_aura_AuraScript);
+
+            void AfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+            {
+                if (Unit* caster = GetCaster())
+                    caster->SetDisableGravity(false);
+            }
+
+            void Register() override
+            {
+                AfterEffectRemove += AuraEffectRemoveFn(spell_dh_fel_rush_aura_AuraScript::AfterRemove, EFFECT_7, SPELL_AURA_MECHANIC_IMMUNITY, AURA_EFFECT_HANDLE_REAL);
```

Comment:
Player often remains stuck in the hover mode. I suggest (if I may of course) using "AURA_EFFECT_HANDLE_SEND_FOR_CLIENT_MASK" instead of AURA_EFFECT_HANDLE_REAL, it solved the problem for me.